mod_modular_vanilla/hooks/skills/effects/disarmed_effect.nut

```diff
@@ -0,0 +1,15 @@
+::ModularVanilla.QueueBucket.VeryLate.push(function() {
+	::ModularVanilla.MH.hook("scripts/skills/effects/disarmed_effect", function(q) {
+		// Part of the actor.interrupt framework
+		// Because this effect doesn't remove shieldwall_effect, we don't want to catch it in auto-interrupt detection
+		// in skill_container.removeByID. Instead we trigger a manual interruption here for offensive effects only.
+		q.onAdded = @(__original) function()
+		{
+			__original();
+			if (!this.isGarbage())
+			{
+				this.getContainer().getActor().interrupt(true, false);
```

Comment:
The concern is that the main thing that Disarmed accomplishes is setting `properties.IsAbleToUseWeaponSkills` to false. There can be other effects which do this. We even added an `actor.isDisarmed()` function in MSU so that modders can extend that to add more effects which may accomplish something which can be considered as "disarmed" for an actor. Therefore, I do not want to assume that the `disarmed_effect` is an edge case that we can ignore in this framework. Instead I'd like a solution which covers all bases.

Your points are fair. But, as described above, I am also not fully satisfied with an interrupt framework which isn't general enough to cover these cases. 

The current implementation isn't perfect. So I'd like more suggestions on how this can be implemented in an elegant way which would achieve the goals of:
- Be general and extensible.
- Be intuitive to use.

Reply:
`actor.isDisarmed()` from MSU should be sufficient. 
Any effect that is bound to the presence of a weapon (Riposte, Spearwall) should check whether this actor is currently disarmed to then remove the effect
My suggestion above of focussing just on the disarmed_effect is not nearly as flexible for this, in comparison.